File: src/main/java/dev/discordmk/neutron/util/AssetGetter.java
package dev.discordmk.neutron.util;

import javax.swing.*;
import java.net.URL;

public class AssetGetter {
    public static ImageIcon getImageIcon(String name) {
        URL imageIconURL = AssetGetter.class.getResource("/assets/icons/" + name + ".png");
        
        if (imageIconURL == null) {
            System.out.println("Error loading icon: " + name);
            return null;
        } else {
            return new ImageIcon(imageIconURL);
        }
    }
}

File: src/main/kotlin/dev/tonimatas/discordmk/util/AssetGetter.kt
package dev.tonimatas.discordmk.util

import javax.swing.ImageIcon

object AssetGetter {
    fun getImageIcon(name: String): ImageIcon {
        val imageIconURL = AssetGetter::class.java.getResource("/assets/icons/$name.png")
        return ImageIcon(imageIconURL)
    }
}